tests/manage/pv_services/test_pvc_evict_ceph_clients.py

```diff
@@ -0,0 +1,114 @@
+import logging
+import pytest
+import random
+import json
+
+from ocs_ci.ocs.ocp import OCP
+from ocs_ci.ocs.exceptions import CommandFailed
+from ocs_ci.framework.testlib import tier2
+from ocs_ci.framework.pytest_customization.marks import skipif_ocs_version
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.node import get_worker_nodes
+from ocs_ci.ocs.resources.pod import get_ceph_tools_pod
+
+logger = logging.getLogger(__name__)
+
+
+@tier2
+@skipif_ocs_version("<4.11")
+class TestPvcEvictCephClients:
+    """
+    Test the ceph-fs PVC (RWX) mount with the following scenarios involving evict ceph-fs clients:
+    1. When two pods tries to mount the same PVC and scheduled on same node
+    2. When two pods tries to mount the same PVC and scheduled on different node
+    """
+
+    @pytest.mark.parametrize(
+        argnames=["node"],
+        argvalues=[
+            pytest.param(
+                "same",
+                marks=[
+                    pytest.mark.bugzilla("1901499"),
+                    pytest.mark.polarion_id("OCS-3985"),
+                ],
+            ),
+            pytest.param(
+                "different",
+                marks=[pytest.mark.polarion_id("OCS-3984")],
+            ),
+        ],
+    )
+    def test_pvc_evict_ceph_clients(self, node, pvc_factory, pod_factory):
+        worker_nodes = get_worker_nodes()
+
+        # create a RWX PVC
+        pvc_obj = pvc_factory(
+            interface=constants.CEPHFILESYSTEM,
+            access_mode=constants.ACCESS_MODE_RWX,
+            status=constants.STATUS_BOUND,
+        )
+
+        # create a pod in particular node
+        selected_node = random.choice(worker_nodes)
+        logger.info(f"creating first pod on node {selected_node}")
+        pod_factory(
+            interface=constants.CEPHFILESYSTEM,
+            pvc=pvc_obj,
+            node_name=selected_node,
+            pod_dict_path=constants.NGINX_POD_YAML,
+            status=constants.STATUS_RUNNING,
+        )
+
+        # fetch the sub-volume path
+        pvc_obj.reload()
+        sub_volume_path = OCP(kind="pv").get(resource_name=pvc_obj.backed_pv)["spec"][
+            "csi"
+        ]["volumeAttributes"]["subvolumePath"]
+
+        # evict ceph-fs clients
+        logger.info("Evicting ceph-fs clients!")
+        ceph_tools_pod_obj = get_ceph_tools_pod()
```

Comment:
Instead of creating obj and from it sending command you can make the code more clean with this:
[https://github.com/red-hat-storage/ocs-ci/blob/39e340e5feb63e201d9694fcf5de5044daaf9d69/ocs_ci/ocs/resources/pod.py#L241](https://github.com/red-hat-storage/ocs-ci/blob/39e340e5feb63e201d9694fcf5de5044daaf9d69/ocs_ci/ocs/resources/pod.py#L241)